gpt4all-backend/CMakeLists.txt

```diff
@@ -71,18 +73,32 @@ foreach(BUILD_VARIANT IN LISTS BUILD_VARIANTS)
                      PROPERTY INTERPROCEDURAL_OPTIMIZATION ${IPO_SUPPORTED})
     endfunction()
 
-    # Add each individual implementation
-    add_library(llamamodel-${BUILD_VARIANT} SHARED
+    # Add each individual implementations
+    add_library(llamamodel-mainline-${BUILD_VARIANT} SHARED
         llamamodel.cpp)
-    prepare_target(llamamodel llama)
+    target_compile_definitions(llamamodel-mainline-${BUILD_VARIANT} PRIVATE
+        LLAMA_VERSIONS=>=3 LLAMA_DATE=999999)
```

Comment:
=>= oh man cmake.. you're kiling me

Reply:
Look at this line: https://github.com/tuxifan/gpt4all/blob/85ee11faec7f35809c6dff8f9b7563284a4e6996/gpt4all-backend/llamamodel.cpp#L360

Reply:
Haha, yup. Looks confusing, is confusing, but does what we need quite flexibly.

Reply:
That conditional should probably be changed to a slightly less cursed variant:

```#if LLAMA_VERSION <= 123456
// ...
#elif LLAMA_VERSION >= 654321
// ...
#endif```

At least then it would be a readily recognizable pattern of tragic stylistic compromise instead of a confusing entirely new way to crush one's hopes and dreams. Would also shrink the cmake side a little.

Pardon the gallows humour, can't help it whenever pre-processor macros seem necessary. ;)